projects/observability/src/shared/components/cartesian/cartesian-chart.component.ts

```diff
@@ -94,17 +99,26 @@ export class CartesianChartComponent<TData> implements OnChanges, OnDestroy {
     }
 
     this.chart = this.chartBuilderService
-      .build<TData>(this.strategy, this.container.nativeElement, this.renderer)
+      .build<TData>(this.strategy, this.container.nativeElement, this.renderer, this.groupId)
       .withSeries(...this.series)
       .withTooltip(
         this.chartTooltipBuilderService.constructTooltip<TData, Series<TData>>(data =>
           this.convertToDefaultTooltipRenderData(data)
         )
       )
+      .withEventListener(ChartEvent.Hover, data => {
+        this.chartSyncService.mouseLocationChange(data, this.groupId);
+      })
       .withEventListener(ChartEvent.Select, selectedData => {
         this.selectionChange.emit(selectedData);
       });
 
+    if (this.groupId !== undefined) {
+      this.chartSyncService?.getLocationChangesForGroup(this.groupId).subscribe(data => {
```

Comment:
use subscription lifecycle to unsubscribe on destroy

Reply:
Actually, do we need this here? The comment about unsubscribing applies no matter where we do it, but the chart already has its group ID so if we move this into the chart code instead of the component, we don't have to expose the showCrosshair method.